homeassistant/components/simplisafe/__init__.py

```diff
@@ -167,20 +158,24 @@ async def async_setup_entry(hass, config_entry):  # noqa: C901
     client_id = await async_get_client_id(hass)
     websession = aiohttp_client.async_get_clientsession(hass)
 
-    try:
-        api = await API.login_via_token(
-            config_entry.data[CONF_TOKEN], client_id=client_id, session=websession
+    async def async_get_api():
+        """Define a helper to get an authenticated SimpliSafe API object."""
+        return await API.login_via_credentials(
+            config_entry.data[CONF_USERNAME],
+            config_entry.data[CONF_PASSWORD],
+            client_id=client_id,
+            session=websession,
         )
+
+    try:
+        api = await async_get_api()
     except InvalidCredentialsError:
-        LOGGER.error("Invalid credentials provided")
-        return False
+        raise ConfigEntryAuthFailed  # pylint: disable=raise-missing-from
```

Comment:
Why should we not use `raise ConfigEntryAuthFailed from err`?

Reply:
Didn't see the need, since `InvalidCredentialsError` can only mean one thing. That said, happy to accommodate!